dds/DCPS/Observer.h

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_OBSERVER_H
+#define OPENDDS_DCPS_OBSERVER_H
+
+#include "RcObject.h"
+#include "Definitions.h"
+#include "SequenceNumber.h"
+
+#include <dds/DdsDcpsCoreC.h>
+#include <dds/DdsDcpsPublicationC.h>
+#include <dds/DdsDcpsSubscriptionC.h>
+#include <dds/DdsDcpsInfoUtilsC.h>
+
+#if !defined (ACE_LACKS_PRAGMA_ONCE)
+#pragma once
+#endif /* ACE_LACKS_PRAGMA_ONCE */
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+class DataSampleElement;
+class ReceivedDataSample;
+class ReceivedDataElement;
+
+class OpenDDS_Dcps_Export Observer
+  : public virtual RcObject
+{
+public:
+  typedef Observer* Ptr;
+  typedef TAO_Objref_Var_T<Observer> Var;
+  typedef TAO_Objref_Out_T<Observer> Out;
+  typedef RcHandle<Observer> Rch;
+
+  typedef unsigned long Event;
+  enum {
```

Comment:
Why not just `enum Event {` here?

Reply:
If we define enum Event {A=1, B=2, C=4}; we can not use Event as a function parameter or returned type, unless we only pass or return A, B, or C by itself. For example:
void register(Observer o, Event e); we can not do this: register(o, A | B); 
because (A | B) is not a valid value in enum Event {A, B, C}.